OsmAnd/src/net/osmand/plus/routing/RouteProvider.java

```diff
@@ -260,11 +260,11 @@ private RouteCalculationResult calculateGpxRoute(RouteCalculationParams routePar
 					routeParams.intermediates, routeParams.ctx, routeParams.leftSide, null, null, routeParams.mode, true);
 		}
 
-		if (routeParams.gpxRoute.useIntermediatePointsRTE) {
+		if (routeParams.gpxRoute.useIntermediatePointsRTE && !routeParams.gpxRoute.followTrack) {
```

Comment:
doesn't look logical, rte should not be used when track is present itself but here there is no such condition at all